sawmill-core/src/main/java/io/logz/sawmill/processors/DeDotProcessor.java

```diff
@@ -0,0 +1,114 @@
+package io.logz.sawmill.processors;
+
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.ProcessResult;
+import io.logz.sawmill.Processor;
+import io.logz.sawmill.annotations.ProcessorProvider;
+import io.logz.sawmill.exceptions.ProcessorConfigurationException;
+import io.logz.sawmill.utilities.JsonUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+@ProcessorProvider(type = "de_dot", factory = DeDotProcessor.Factory.class)
+public class DeDotProcessor implements Processor {
+    private String fieldSplit;
+    private static final Logger logger = LoggerFactory.getLogger(DeDotProcessor.class);
+
+    public DeDotProcessor(String fieldSplit) {
+        this.fieldSplit = fieldSplit;
+    }
+
+    @Override
+    public ProcessResult process(Doc doc) {
+        Map<String,Object> docAsMap = doc.getSource();
+        deDotMap(docAsMap);
+        return ProcessResult.success();
+    }
+
+    private void deDotMap(Map<String, Object> docAsMap) {
+        //Creating a copy of the Map in this level , and also in each level in the recursive level is required
+        //in order to avoid ConcurrentModificationException , because in our case we want to replace the key thus actually we remove & add .
+        //add is not supported while iterating , and the end result is to create a copy of the map.
+        Map<String,Object> mapClone = new LinkedHashMap<>();
+        deDotInner(docAsMap,mapClone);
+        docAsMap.clear();
+        docAsMap.putAll(mapClone);
+    }
+
+    private void deDotInner(Map<String, Object> docAsMap, Map<String,Object> mapClone) {
```

Comment:
```suggestion
    private Map<String,Object> deDotMap(Map<String, Object> orig) {
```

Reply:
Since we have to invoke the method with a clone of the map from the outer recursive function , and this is also actually the return value , there is less sense in also returning it in the method body....